test/Worker.Extensions.Tests/Table/TableObjectConverterTests.cs

```diff
@@ -0,0 +1,171 @@
+        }
+
+        [Fact]
+        public async Task ConvertAsync_CollectionPocoEntity_WithRowKey_ReturnsSuccess()
+        {
+            object source = GrpcTestHelper.GetTestGrpcModelBindingData(TableTestHelper.GetTableEntityBinaryData(), "AzureStorageTables");
+            var context = new TestConverterContext(typeof(IEnumerable<MyEntity>), source);
+            var mockResponse = new Mock<Response>();
+            var tableClient = new Mock<TableClient>();
+
+            tableClient
+                .Setup(c => c.GetEntityAsync<TableEntity>(It.IsAny<string>(), It.IsAny<string>(), null, default))
+                .ReturnsAsync(Response.FromValue(new TableEntity(It.IsAny<string>(), It.IsAny<string>()), mockResponse.Object));
+
+            _mockTableServiceClient
+                .Setup(c => c.GetTableClient(Constants.TableName))
+                .Returns(tableClient.Object);
+
+            var expectedOutput = Page<TableEntity>.FromValues(new List<TableEntity> { new TableEntity("partitionKey", "rowKey") }, continuationToken: null, mockResponse.Object);
+
+            tableClient
+                .Setup(c => c.QueryAsync<TableEntity>(It.IsAny<string>(), null, null, default))
+                .Returns(AsyncPageable<TableEntity>.FromPages(new List<Page<TableEntity>> { expectedOutput }));
+
+            var conversionResult = await _tableConverter.ConvertAsync(context);
+
+            Assert.Equal(ConversionStatus.Succeeded, conversionResult.Status);
+        }
+
+        [Fact]
+        public async Task ConvertAsync_ModelBindingData_Null_ReturnsUnhandled()
+        {
+            var context = new TestConverterContext(typeof(IEnumerable<MyEntity>), null);
+
+            var conversionResult = await _tableConverter.ConvertAsync(context);
+
+            Assert.Equal(ConversionStatus.Unhandled, conversionResult.Status);
+        }
+
+        [Fact]
+        public async Task ConvertAsync_SingleTableEntity_NullTargetType_ReturnsFailed()
+        {
+            object source = GrpcTestHelper.GetTestGrpcModelBindingData(TableTestHelper.GetTableEntityBinaryData(), "AzureStorageTables");
+            var context = new TestConverterContext(null, source);
+            var mockResponse = new Mock<Response>();
+            var tableClient = new Mock<TableClient>();
+
+            tableClient
+                .Setup(c => c.GetEntityAsync<TableEntity>(It.IsAny<string>(), It.IsAny<string>(), null, default))
+                .ReturnsAsync(Response.FromValue(new TableEntity(It.IsAny<string>(), It.IsAny<string>()), mockResponse.Object));
+
+            _mockTableServiceClient
+                .Setup(c => c.GetTableClient(Constants.TableName))
+                .Returns(tableClient.Object);
+
+            var conversionResult = await _tableConverter.ConvertAsync(context);
+
+            Assert.Equal(ConversionStatus.Failed, conversionResult.Status);
+        }
+    }
+
+    class MyEntity
```

Comment:
In the tests, make sure to cover both scenarios: use the built-in `TableEntity` as well as a custom class that implements `ITableEntity` with additional properties.